File: ruby/strategy/DuckSimulator/client/redhead_duck.rb
require "./client/duck"
require './behavior/fly_behavior'
require './behavior/quack_behavior'

class RedheadDuck < Duck

  def initialize(duck_name)
    super(duck_name)
    @fly_behavior = FlyWithWings.new(duck_name)
    @quack_behavior = Squeak.new(duck_name)
  end

end

File: javascript/strategy/DuckSimulator/client/RedheadDuck.js
import {Duck} from "./Duck.js";
import {FlyWithWings} from '../behavior/FlyBehavior.js'
import {Squeak} from '../behavior/QuackBehavior.js'

export class RedheadDuck extends Duck {

    constructor(duck_name) {
        super(duck_name)
        this.fly_behavior = new FlyWithWings(duck_name)
        this.quack_behavior = new Squeak(duck_name)
    }
}